src/main/java/Parser.java

```diff
@@ -0,0 +1,154 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Parser {
+
+    private String raw;
+    List<String> args;
+    String type;
+    String remainingArgs;
+    List<String> argsWithDate;
+
+    /**
+     * Constructor for Parser.
+     *
+     * @param move string that user types in
+     */
+    public Parser(String move) {
+        raw = move;
+        args = new LinkedList<>(Arrays.asList(move.split(" ")));
+        type = args.get(0);
+        args.remove(0);
+        remainingArgs = String.join(" ", args);
+    }
+
+    /**
+     * Gets for type of action.
+     *
+     * @return type of action
+     */
+    public String getType() {
+        if (args.size() == 0) {
+            return raw;
+        } else {
+            return type;
+        }
+    }
+
+    /**
+     * Gets for Task Number from action.
+     *
+     * @return Task Number
+     * @throws DukeException if task number incorrect
```

Comment:
Correct format for documenting an exception. But you should capitalise your **If**